controllers/fenceagentsremediation_controller.go

```diff
@@ -253,20 +254,31 @@ func (r *FenceAgentsRemediationReconciler) updateStatus(ctx context.Context, far
 	return nil
 }
 
-// updateConditions updates the status conditions of a FenceAgentsRemediation object based on the provided ProcessingChangeReason.
-// return an error if an unknown ProcessingChangeReason is provided
-func updateConditions(reason v1alpha1.ProcessingChangeReason, currentConditions *[]metav1.Condition, log logr.Logger) error {
+// updateConditions updates the status conditions of a FenceAgentsRemediation object based on the provided ConditionsChangeReason.
+// return an error if an unknown ConditionsChangeReason is provided
+func updateConditions(reason v1alpha1.ConditionsChangeReason, currentConditions *[]metav1.Condition, log logr.Logger) error {
 
 	var (
 		processingConditionStatus, fenceAgentActionSucceededConditionStatus, succeededConditionStatus metav1.ConditionStatus
 		conditionMessage                                                                              string
 	)
-	// All the ProcessingChangeReasons can happen one after another
-	// RemediationStarted will always be the first reason.
+	// RemediationFinishedNodeNotFound and RemediationInterruptedByNHC reasons can happen at any time the Reconcile runs
+	// Except these two reasons, there are another three reasons that can only happen one after another
+	// RemediationStarted will always be the first reason (out of these three)
 	// FenceAgentSucceeded can only happen after RemediationStarted happened
-	// RemediationFinished can only happen after FenceAgentSucceeded happened
+	// RemediationFinishedSuccessfully can only happen after FenceAgentSucceeded happened
 
 	switch reason {
+	case v1alpha1.RemediationFinishedNodeNotFound, v1alpha1.RemediationInterruptedByNHC:
+		processingConditionStatus = metav1.ConditionFalse
+		fenceAgentActionSucceededConditionStatus = metav1.ConditionFalse
+		succeededConditionStatus = metav1.ConditionFalse
+		if reason == v1alpha1.RemediationFinishedNodeNotFound {
```

Comment:
IMHO a switch makes more sense here, which errors when a reason isn't handled. Prevents forgetting this part when adding another reason to the case statement above.

Reply:
Good point